src/main/scala/nodes/images/Pooler.scala

```diff
@@ -67,3 +67,21 @@ class Pooler(
     ChannelMajorArrayVectorizedImage(patch, ImageMetadata(numPoolsX, numPoolsY, numChannels))
   }
 }
+
+object Pooler {
+
+  def sumPooler(in: DenseVector[Double]): Double = {
```

Comment:
Hmm.. ".sum" is deprecated, and on *my* laptop - `sum(x)` (well, with an appropriate change to "var sum=0.0" above), goes 2x faster than this while loop.